subcrates/gelatin/src/application.rs

```diff
@@ -97,9 +99,13 @@ impl Application {
 								..
 							},
 						..
-					} => {
-						close_requested = true;
-					}
+					} => match windows.get(&control_flow_source) {
+						Some(window) if window.fullscreen() => {
+							window.set_fullscreen(false);
+							window.process_event(event);
+						}
+						_ => close_requested = true,
+					},
```

Comment:
As I mentioned in my previous comment this should instead be handled in `picture_widget.rs` from where the fullscreen mode is being set.

Reply:
I removed line 104. Is this what you meant?

Reply:
The minimal change that's somewhat clean is to completely remove the part from `aplication.rs` that sets `close_requested` to true when escape is pressed. This is something that the gui framework should not do anyways. Instead listen for the espace key event in the `picture_widget`. I don't think we should add a customizable action binding for this so don't use the triggered macro, just hard-code a `VirtualKeyCode::Escape` check at `EventKind::KeyInput`

Reply:
Oh now I see, then it's absolutely non-trivial how the picture_widget will make the application quit if the window wasn't full-screen. I'm going to add something to the master that kinda solves this which you can then merge back to your branch and use.

Reply:
@Aloso the master branch now has a public `request_exit()` function in the application.rs file. You can use that from the picture_widget to initiate an exit when the window wasn't full-screen.

Reply:
Thanks, that helped a lot!